src/lib/interfaces/request.ts

```diff
@@ -4,6 +4,11 @@ interface HttpRequest {
   headers?: object
   data?: {
     fields?: object
+    // TODO: This is currently really confusing: We are reusing the
+    // httpRequest interface which is taylored for cts and therefore
+    // expects only a fields property. But for a tag creation it
+    // should instead expect a name property.
+    name?: string
```

Comment:
IIs there anything we can do about this? Maybe a discriminated union so we can share the same properties. but know it's actual sub-type?

Is it always mutually exclusive to have either `fields` or `name` in data?

Reply:
Yes, it is mutually exclusive.

I will have a look at what a discriminated union is and if we can apply it here :)

Reply:
But can't a content type have a name as well? 🤔

Reply:
Good point. :thinking: I will try and understand why and how this worked before without the `name` property as part of the interface.

Reply:
Okay, it seems like I just did not understand the typescript validation properly. Typescript complained, because the `toApi` method of the tag entity returns a type that will never have a `field` property, so I could not assign it to an optional type `data` which when used requires one of its optional properties to be part of the interface. So there of course actually is in general a `name` property for content type requests, which makes all of this less confusing. `fields` is just another additional optional property. I would suggest then to just leave it like this, but also add the optional parameters `description` and `displayField`.

Reply:
I updated it accordingly.